sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java

```diff
@@ -89,16 +91,20 @@ private void getCachedOffers() {
 				OfferListUtil.splitOffersByType(cachedOfferList.getOffers(), this.earnList, this.spendList);
 			}
 		}
-		setOfferLists();
+		setCachedOfferLists();
 	}
 
-	private void setOfferLists() {
+	private void setCachedOfferLists() {
 		if (this.view != null) {
 			this.view.setEarnList(earnList);
 			this.view.setSpendList(spendList);
 
-			setEarnEmptyViewState();
-			setSpendEmptyViewState();
+			if (!earnList.isEmpty()) {
```

Comment:
why checking !earnList.isEmpty()? and not calling updateEarnTitle() in anycase? same for !spendList.isEmpty()?